src/rust/engine/process_execution/src/cache.rs

```diff
@@ -100,6 +102,7 @@ impl CommandRunner {
             execute_response,
             vec![],
             context.workunit_store,
+            Platform::Linux, //TODO this is incorrect and will be fixed in a follow-up PR
```

Comment:
Just to check, can you sketch out how you're planning to fix it before merging?

Reply:
https://github.com/pantsbuild/pants/pull/9198 is the draft PR that fixes this, that I need to give a little bit more attention to before flipping it out of draft mode.